### PR TITLE
Avoid mutating the environment with RUBYOPT

### PR DESCRIPTION
**Pull request by vinistock**
If we simply assign `env = process.env`, then when we change keys in `env` we're mutating `process.env` as well - which leaks across different extensions.

This means that we append `RUBYOPT=--yjit` in the process environment and it ends up applying to other extensions that spawn Ruby processes, which may lead to issues.

The fix is to simply create a clone of the environment, which we can do by destructuring it inside a new object.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "ruby-lsp",
   "displayName": "Ruby LSP",
   "description": "VS Code plugin for connecting with the Ruby LSP",
-  "version": "0.1.3",
+  "version": "0.1.4",
```
Comment (KaanOzkan) [inline package.json:5]: I'm assuming the version bump is intentional. Maybe could be in its own commit.

**Review by st0012 — APPROVED** [review]
